### ISSUE TITLE
Drag and Drop or Pasting images: upload is successful, but a black square remains on editor

### ISSUE THREAD

**Issue by feliperaul** [galetahub/ckeditor#836]
First, thanks for all the maintainers for this great gem.

I didn't find any reference in the README about support for Drag and Drop or Pasting images, so I assumed this wouldn't be supported.

I only found out it was supported after hitting CMD V intuitively; since this is a extremely useful feature, I'm willing to write a README improvement, but I need to get it 100% working first.

What happens as soon as I drag an image or paste it is this:

- I get a success warning inside the editor, but only a black square remains inline:

![image](https://user-images.githubusercontent.com/743879/53811252-ae357180-3f37-11e9-822d-cefbbd48d9db.png)

Double clicking that image, I see that it has a URL of `data:image/gif;base64,R0lGODlhDgAOAIAAAAAAAP///yH5BAAAAAAALAAAAAAOAA4AAAIMhI+py+0Po5y02qsKADs=`

But the funny thing is that the image upload WAS successful. If after double clicking on the black square I click "Locate on server", I get a popup with my uploaded images and the image is right there.

Rails console also confirms that the upload was successful.

Checking the Chrome network tab, as soon as I hit CMD+V or drag the image I get this:

![image](https://user-images.githubusercontent.com/743879/53811217-9eb62880-3f37-11e9-8ad6-ef7ec6541a8c.png)

So there are 2 get requets and one POST request; the POST was sent to `ckeditor/pictures?&responseType=json` and the response was `{"uploaded":1,"fileName":"image.png","url":"/uploads/editor/pictures/18f92a7089c4/image.png"}`

There are no JS errors in the console. 

As far as I could tell, this feature was added to the gem in #604 , but looking at the source code I only see controller changes. The pull request states that:

- _Uploading pasted and dragged images was introduced in CKEditor 4.5 (provided through an optional Upload Image)_ , but I don't see `config.extraPlugins = 'uploadimage'` defined anywhere, not in my code nor in the gem's code (which I thought was necessary by reading [CKEDITOr's docs](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_file_upload.html)). But adding it myself to my js config file didn't fix the issue. 
- _And you have to define this in config.js: config.imageUploadUrl = "/ckeditor/pictures?responseType=json"_ - but I didn't have to do any of this, what I have is `config.filebrowserImageUploadUrl = "/ckeditor/pictures?";` 

My point is: this is such a great feature, and it seems to _almost_ be working out of the box; all the difficult parts are already in place (it's detecting the CMD+V or drag and drop, uploading the file, and returning a JSON with the image location to the browser), so what can I do to have this 100% working? I  will be glad  to issue a pull request with a fix in the README because I think this feature  (once working) should be mentioned!

**Comment by feliperaul** [comment]
According to [this issue](https://stackoverflow.com/a/34930418/1290457), this behavior occurs when `LoopIndex Track Changes` plugin is enabled, but I don't see it anywhere in the build.

**Comment by feliperaul** [comment]
@patricklindsay Patrick, even tough I see `uploadimage` in  the build-config, I don't see it anywhere in the repo. Could you confirm if the gem supports the paste / drag and drop feature, and if it's working for you on master branch?

**Comment by patricklindsay** [comment]
I'll check if this is working on master but not sure if this functionality was ever supported.

**Comment by feliperaul** [comment]
If it’s not working, I think we are very close to make it work, since the upload already works on CMD V / drag and drop and the server returns a correctly formatted json. 

As far as I could tell it seems it’s not 100% there because the ckeditor build is missing the uplodimage plugin, which is weird, because it’s part of the default build and it’s also listed on the build-config file, but I might be wrong.

**Comment by feliperaul** [comment]
@patricklindsay Hey Patrick, did you ever manage to take a look on this? It's so close to be working but I really don't know how to proceed to fix the JS problem.

**Comment by patricklindsay** [comment]
@feliperaul I can confirm this works on

**Comment by patricklindsay** [comment]
@feliperaul I can confirm this works on `ckeditor (4.3.0)` in production using `Rails 5.2.x`. The only change required is (as you said) to add a `?` to the upload URL, i.e.

```
  <%-
     pictures_path = Ckeditor::Engine.routes.url_helpers.pictures_path
  %>
  /* Filebrowser routes */
...
  config.filebrowserImageUploadUrl = "<%= pictures_path %>?"
...
```
Check here for why the `responseType=json` isn't required - 
https://github.com/ckeditor/ckeditor4/blob/4.11.1/plugins/filebrowser/plugin.js#L140-L151

You also do not need to add `uploadimage` as an extra plugin as it's included in `ckeditor (4.3.0)`

Can't see why yours wasn't working. If you want, create a sample app with the issue and I'll try debug it, although make sure it's running on the same gem version & Rails version as mentioned above.
